Firestore/core/src/firebase/firestore/api/settings.cc

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Firestore/core/src/firebase/firestore/api/settings.h"
+
+#include "Firestore/core/src/firebase/firestore/util/hashing.h"
+
+namespace firebase {
+namespace firestore {
+namespace api {
+
+size_t Settings::Hash() const {
+  return util::Hash(host_, ssl_enabled_, persistence_enabled_,
+                    timestamps_in_snapshots_enabled_, cache_size_bytes_);
+}
+
+bool operator==(const Settings& lhs, const Settings& rhs) {
```

Comment:
I ended up not needing operator==, so I could drop this (and Hash() presumably)...  or is it convention to always implement them?

Reply:
It may not be useful here but it will be useful for public C++ so let's leave it (given that you already wrote it).